### DISCUSSION TITLE
Login

### DISCUSSION THREAD

**Discussion by FuckingSherlock** [tginfo/sticker-importer#8] [Ideas]
It is necessary to develop a login through a browser or application, as in all services.  It's more convenient and you don't need to get a code.  And safer.

**Comment by Sominemo** [comment]
Importer can't use browser login and has to pretend to be the official app instead, because VK doesn't give access to its sticker store to third-party developers, especially for such purposes as cloning the packs.

You can still use the main functionality (importing regular sticker packs) without signing in if you don't trust the app.

**Comment by FuckingSherlock** [comment]
It is true that it can also be used without login, but it is very inconvenient not to be able to copy the emojis corresponding to the sticker. My distrust of the application is due to the fact that after trying to log in, I received a notification from VK that an attempt was made to log in from Moscow from an IOS device. Although I use Android and am not in Moscow. 
--
Отправлено из Mail.ru для Android суббота, 27 мая 2023г., 17:04 +03:00 от Sominemo  ***@***.*** :

**Reply by Sominemo** [reply]
> it is very inconvenient not to be able to copy the emojis corresponding to the sticker

Sorry to hear that, the emoji are pulled from VK API and auth is required for that.


> from Moscow

If you are from Ukraine, the app uses VK's official proxies. If you are from Russia, it's probably just VK's GeoIP being bad. For example, cellular operators often issue their consumers IPs in Moscow even if they are in some other region.


> from an IOS device

The app needs to pretend to be the iOS client to access the data it needs; Android's app ID was used initially but had some data missing for odd reasons.


All code related to sign in can be inspected in:
https://github.com/tginfo/sticker-importer/tree/master/lib/services/connection
https://github.com/Sominemo/VkGet

If you don't trust the pre-built binaries I provide, you can try building the app itself, it's relatively straightforward (but time consuming if you don't already have Android SDK installed). There's nothing better I can suggest, VK's API became very hostile to third-patry developers in recent years, and this is the only way.

**Comment by FuckingSherlock** [comment]
А я вот думаю нафиг мы по английски говорим?... 
Вобщем, понимаю. Апи вк - впринципе та еще хрень. 
--
Отправлено из Mail.ru для Android суббота, 27 мая 2023г., 17:18 +03:00 от Sominemo  ***@***.*** :
